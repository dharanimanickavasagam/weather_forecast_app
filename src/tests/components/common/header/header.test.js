import React from "react";
import { shallow } from "enzyme";
import Header from "../../../../components/common/header";

let wrapper, mockName, mockVariant;

beforeEach(() => {
  mockName = "mockerName";
  mockVariant = "h6";
  wrapper = shallow(<Header name={mockName} variant={mockVariant} />);
});

describe("it renders component Header", () => {
  test("it should render the component with the passed props", () => {
    expect(
      wrapper.find("WithStyles(ForwardRef(Typography))").prop("variant")
    ).toEqual(mockVariant);

    expect(wrapper.find("WithStyles(ForwardRef(Typography))").text()).toBe(
      mockName
    );
    expect(wrapper).toMatchSnapshot();
  });
});
