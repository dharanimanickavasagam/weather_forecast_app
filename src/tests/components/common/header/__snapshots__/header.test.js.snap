// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders component Header it should render the component with the passed props 1`] = `
<WithStyles(ForwardRef(AppBar))
  position="static"
>
  <WithStyles(ForwardRef(Toolbar))>
    <WithStyles(ForwardRef(Typography))
      variant="h6"
    >
      mockerName
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Toolbar))>
</WithStyles(ForwardRef(AppBar))>
`;
